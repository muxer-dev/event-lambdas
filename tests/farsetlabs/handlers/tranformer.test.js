const { prefix, resolved } = require("../test-utils");
const { getFromS3 } = require(`${prefix}/node_modules/aws-lambda-data-utils`);
const { uploadTo } = require(`${prefix}/node_modules/@muxer/lambda-utils`);
const transformer = require(`${prefix}/handlers/transformer`);

const requireData = filename =>
  JSON.stringify(require(`./test-data/${filename}.json`)); // eslint-disable-line global-require

const emptyEventsList = requireData("events-payload-empty");
const invalidEventsList = requireData("events-payload-invalid");
const incompleteEventsList = requireData("events-payload-incomplete");
const populatedEventsList = requireData("events-payload-populated");

const context = null;
let event;

describe("Farsetlabs transformer", function() {
  beforeEach(function() {
    jest.clearAllMocks();
    event = null;
    uploadTo.mockImplementation(resolved({ key: "path/to/new/file.json" }));
  });

  it("is a lambda handler", function() {
    expect(typeof transformer).toBe("object");
    expect(typeof transformer.transform).toBe("function");
  });

  describe("when no new file is added to the bucket", function() {
    beforeEach(function() {
      event = {
        Records: []
      };
    });

    it("does not return an error and returns empty message", function(done) {
      transformer.transform(event, context, function(err, response) {
        expect(err).toBe(null);
        expect(response).toEqual({ message: [] });
        done();
      });
    });

    it("does not uploads anything", function(done) {
      transformer.transform(event, context, function() {
        expect(uploadTo).toHaveBeenCalledTimes(0);
        done();
      });
    });
  });

  describe("when a new file is added to the bucket", function() {
    beforeEach(function() {
      event = {
        Records: [
          {
            s3: {
              bucket: "test-muxer-bucket",
              object: { Key: "path/to/events.json" }
            }
          }
        ]
      };
    });

    describe("which contains no events", function() {
      beforeEach(function() {
        getFromS3.mockImplementation(resolved({ Body: emptyEventsList }));
      });

      it("does not return an error and returns message", function(done) {
        transformer.transform(event, context, function(err, response) {
          expect(err).toBe(null);
          expect(response).toEqual({ message: ["path/to/new/file.json"] });
          done();
        });
      });

      it("uploads empty events list", function(done) {
        transformer.transform(event, context, function() {
          expect(uploadTo).toHaveBeenCalledTimes(1);
          expect(uploadTo).toHaveBeenCalledWith(
            "muxer-transformed-events",
            expect.any(Function),
            []
          );
          done();
        });
      });
    });

    describe("which contains an invalid event", function() {
      beforeEach(function() {
        getFromS3.mockImplementation(resolved({ Body: invalidEventsList }));
      });

      it("returns an error", function(done) {
        transformer.transform(event, context, function(err, response) {
          expect(err).toEqual(
            new TypeError("Cannot read property 'dateTime' of undefined")
          );
          expect(response).toBe(null);
          done();
        });
      });

      it("does not upload data", function(done) {
        transformer.transform(event, context, function() {
          expect(uploadTo).not.toHaveBeenCalled();
          done();
        });
      });
    });

    describe("which contains incomplete events", function() {
      beforeEach(function() {
        jest.spyOn(global.console, "log").mockImplementation(() => {});
        getFromS3.mockImplementation(resolved({ Body: incompleteEventsList }));
      });

      it("does not return an error and returns message", function(done) {
        transformer.transform(event, context, function(err, response) {
          expect(err).toBe(null);
          expect(response).toEqual({ message: ["path/to/new/file.json"] });
          done();
        });
      });

      it("uploads transformed events list, without incomplete events", function(done) {
        transformer.transform(event, context, function() {
          expect(uploadTo).toHaveBeenCalledTimes(1);
          expect(uploadTo.mock.calls[0][2].length).toBe(4);
          expect(uploadTo.mock.calls).toMatchSnapshot();
          done();
        });
      });

      it("outputs warning", function(done) {
        /* eslint-disable no-console */
        transformer.transform(event, context, function() {
          expect(console.log).toHaveBeenCalledTimes(1);
          expect(console.log).toHaveBeenCalledWith(
            "WARNING: some events generated were not valid!"
          );
          done();
        });
      });
    });

    describe("which contains complete events", function() {
      beforeEach(function() {
        getFromS3.mockImplementation(resolved({ Body: populatedEventsList }));
      });

      it("does not return an error and returns message", function(done) {
        transformer.transform(event, context, function(err, response) {
          expect(err).toBe(null);
          expect(response).toEqual({ message: ["path/to/new/file.json"] });
          done();
        });
      });

      it("uploads transformed events list", function(done) {
        transformer.transform(event, context, function() {
          expect(uploadTo).toHaveBeenCalledTimes(1);
          expect(uploadTo.mock.calls[0][2].length).toBe(5);
          expect(uploadTo.mock.calls).toMatchSnapshot();
          done();
        });
      });
    });
  });

  describe("when multiple new files are added to the bucket", function() {
    beforeEach(function() {
      event = {
        Records: [
          {
            s3: {
              bucket: "test-muxer-bucket",
              object: { Key: "path/to/events-1.json" }
            }
          },
          {
            s3: {
              bucket: "test-muxer-bucket",
              object: { Key: "path/to/events-2.json" }
            }
          }
        ]
      };

      getFromS3
        .mockReturnValueOnce(resolved({ Body: populatedEventsList })())
        .mockReturnValueOnce(resolved({ Body: populatedEventsList })());
    });

    it("does not return an error and returns messages", function(done) {
      transformer.transform(event, context, function(err, response) {
        expect(err).toBe(null);
        expect(response).toEqual({
          message: ["path/to/new/file.json", "path/to/new/file.json"]
        });
        done();
      });
    });

    it("uploads transformed events lists", function(done) {
      transformer.transform(event, context, function() {
        expect(uploadTo).toHaveBeenCalledTimes(2);
        expect(uploadTo.mock.calls[0][2].length).toBe(5);
        expect(uploadTo.mock.calls[1][2].length).toBe(5);
        expect(uploadTo.mock.calls).toMatchSnapshot();
        done();
      });
    });
  });
});
