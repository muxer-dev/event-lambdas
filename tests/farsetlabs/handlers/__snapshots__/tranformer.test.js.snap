// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Farsetlabs transformer when a new file is added to the bucket which contains complete events uploads transformed events list 1`] = `
Array [
  Array [
    "muxer-transformed-events",
    [Function],
    Array [
      Object {
        "created_at": "2018-05-21T10:21:12.000Z",
        "description": "https://www.meetup.com/CodeCoop-NI",
        "last_updated": "2018-05-21T10:33:48.120Z",
        "name": "Code Co-op NI",
        "source_data": Object {
          "id": "4b471u35laf5kgeegsjjku3lu8_20181018T180000Z",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 7200000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-18T20:00:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-18T18:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
      Object {
        "created_at": "2017-09-19T08:01:05.000Z",
        "description": "http://www.farsetlabs.org.uk/events/coderdojo.html",
        "last_updated": "2017-09-19T08:01:06.267Z",
        "name": "Farset Labs Dojo 2017",
        "source_data": Object {
          "id": "0mdpdpn59po7htqvf7o9bercmt_20181019T170000Z",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 8400000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-19T19:20:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-19T17:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
      Object {
        "created_at": "2018-08-10T19:01:39.000Z",
        "description": "Halloween Cyber Security Challenge (Because vulnerability is scary)
Code Co-op: https://www.meetup.com/CodeCoop-NI/
Contact Zoe or Art for more details.",
        "last_updated": "2018-08-10T19:04:21.794Z",
        "name": "Code Co-op: Halloween Cyber-Security Challenge",
        "source_data": Object {
          "id": "316kag8mfr56qt58k3guo4ejsr",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 10800000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-30T21:00:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-30T18:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
      Object {
        "created_at": "2013-10-29T11:11:12.000Z",
        "description": " ",
        "last_updated": "2016-09-25T16:33:30.395Z",
        "name": "Town Hall",
        "source_data": Object {
          "id": "sh66lg4mn4uj77a3nk7odiep14_20181105T190000Z",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 7200000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-11-05T21:00:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-11-05T19:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
      Object {
        "created_at": "2018-10-05T07:04:40.000Z",
        "description": "Details TBC",
        "last_updated": "2018-10-05T07:04:43.673Z",
        "name": "Design Science [Belfast Design Week]",
        "source_data": Object {
          "id": "0f6joijanqo2do55253lp6a6hh",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 7200000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-11-08T21:00:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-11-08T19:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
    ],
  ],
]
`;

exports[`Farsetlabs transformer when a new file is added to the bucket which contains incomplete events uploads transformed events list, without incomplete events 1`] = `
Array [
  Array [
    "muxer-transformed-events",
    [Function],
    Array [
      Object {
        "created_at": "2018-05-21T10:21:12.000Z",
        "description": "https://www.meetup.com/CodeCoop-NI",
        "last_updated": "2018-05-21T10:33:48.120Z",
        "name": "Code Co-op NI",
        "source_data": Object {
          "id": "4b471u35laf5kgeegsjjku3lu8_20181018T180000Z",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 7200000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-18T20:00:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-18T18:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
      Object {
        "created_at": "2017-09-19T08:01:05.000Z",
        "description": "http://www.farsetlabs.org.uk/events/coderdojo.html",
        "last_updated": "2017-09-19T08:01:06.267Z",
        "name": "Farset Labs Dojo 2017",
        "source_data": Object {
          "id": "0mdpdpn59po7htqvf7o9bercmt_20181019T170000Z",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 8400000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-19T19:20:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-19T17:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
      Object {
        "created_at": "2018-08-10T19:01:39.000Z",
        "description": "Halloween Cyber Security Challenge (Because vulnerability is scary)
Code Co-op: https://www.meetup.com/CodeCoop-NI/
Contact Zoe or Art for more details.",
        "last_updated": "2018-08-10T19:04:21.794Z",
        "name": "Code Co-op: Halloween Cyber-Security Challenge",
        "source_data": Object {
          "id": "316kag8mfr56qt58k3guo4ejsr",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 10800000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-30T21:00:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-30T18:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
      Object {
        "created_at": "2018-10-05T07:04:40.000Z",
        "description": "Details TBC",
        "last_updated": "2018-10-05T07:04:43.673Z",
        "name": "Design Science [Belfast Design Week]",
        "source_data": Object {
          "id": "0f6joijanqo2do55253lp6a6hh",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 7200000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-11-08T21:00:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-11-08T19:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
    ],
  ],
]
`;

exports[`Farsetlabs transformer when multiple new files are added to the bucket uploads transformed events lists 1`] = `
Array [
  Array [
    "muxer-transformed-events",
    [Function],
    Array [
      Object {
        "created_at": "2018-05-21T10:21:12.000Z",
        "description": "https://www.meetup.com/CodeCoop-NI",
        "last_updated": "2018-05-21T10:33:48.120Z",
        "name": "Code Co-op NI",
        "source_data": Object {
          "id": "4b471u35laf5kgeegsjjku3lu8_20181018T180000Z",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 7200000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-18T20:00:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-18T18:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
      Object {
        "created_at": "2017-09-19T08:01:05.000Z",
        "description": "http://www.farsetlabs.org.uk/events/coderdojo.html",
        "last_updated": "2017-09-19T08:01:06.267Z",
        "name": "Farset Labs Dojo 2017",
        "source_data": Object {
          "id": "0mdpdpn59po7htqvf7o9bercmt_20181019T170000Z",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 8400000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-19T19:20:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-19T17:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
      Object {
        "created_at": "2018-08-10T19:01:39.000Z",
        "description": "Halloween Cyber Security Challenge (Because vulnerability is scary)
Code Co-op: https://www.meetup.com/CodeCoop-NI/
Contact Zoe or Art for more details.",
        "last_updated": "2018-08-10T19:04:21.794Z",
        "name": "Code Co-op: Halloween Cyber-Security Challenge",
        "source_data": Object {
          "id": "316kag8mfr56qt58k3guo4ejsr",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 10800000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-30T21:00:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-30T18:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
      Object {
        "created_at": "2013-10-29T11:11:12.000Z",
        "description": " ",
        "last_updated": "2016-09-25T16:33:30.395Z",
        "name": "Town Hall",
        "source_data": Object {
          "id": "sh66lg4mn4uj77a3nk7odiep14_20181105T190000Z",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 7200000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-11-05T21:00:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-11-05T19:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
      Object {
        "created_at": "2018-10-05T07:04:40.000Z",
        "description": "Details TBC",
        "last_updated": "2018-10-05T07:04:43.673Z",
        "name": "Design Science [Belfast Design Week]",
        "source_data": Object {
          "id": "0f6joijanqo2do55253lp6a6hh",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 7200000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-11-08T21:00:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-11-08T19:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
    ],
  ],
  Array [
    "muxer-transformed-events",
    [Function],
    Array [
      Object {
        "created_at": "2018-05-21T10:21:12.000Z",
        "description": "https://www.meetup.com/CodeCoop-NI",
        "last_updated": "2018-05-21T10:33:48.120Z",
        "name": "Code Co-op NI",
        "source_data": Object {
          "id": "4b471u35laf5kgeegsjjku3lu8_20181018T180000Z",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 7200000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-18T20:00:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-18T18:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
      Object {
        "created_at": "2017-09-19T08:01:05.000Z",
        "description": "http://www.farsetlabs.org.uk/events/coderdojo.html",
        "last_updated": "2017-09-19T08:01:06.267Z",
        "name": "Farset Labs Dojo 2017",
        "source_data": Object {
          "id": "0mdpdpn59po7htqvf7o9bercmt_20181019T170000Z",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 8400000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-19T19:20:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-19T17:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
      Object {
        "created_at": "2018-08-10T19:01:39.000Z",
        "description": "Halloween Cyber Security Challenge (Because vulnerability is scary)
Code Co-op: https://www.meetup.com/CodeCoop-NI/
Contact Zoe or Art for more details.",
        "last_updated": "2018-08-10T19:04:21.794Z",
        "name": "Code Co-op: Halloween Cyber-Security Challenge",
        "source_data": Object {
          "id": "316kag8mfr56qt58k3guo4ejsr",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 10800000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-30T21:00:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-10-30T18:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
      Object {
        "created_at": "2013-10-29T11:11:12.000Z",
        "description": " ",
        "last_updated": "2016-09-25T16:33:30.395Z",
        "name": "Town Hall",
        "source_data": Object {
          "id": "sh66lg4mn4uj77a3nk7odiep14_20181105T190000Z",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 7200000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-11-05T21:00:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-11-05T19:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
      Object {
        "created_at": "2018-10-05T07:04:40.000Z",
        "description": "Details TBC",
        "last_updated": "2018-10-05T07:04:43.673Z",
        "name": "Design Science [Belfast Design Week]",
        "source_data": Object {
          "id": "0f6joijanqo2do55253lp6a6hh",
          "name": "farsetlabs-calendar",
        },
        "times": Object {
          "duration": 7200000,
          "end": Object {
            "timezone": "Europe/London",
            "utc": "2018-11-08T21:00:00.000Z",
          },
          "start": Object {
            "timezone": "Europe/London",
            "utc": "2018-11-08T19:00:00.000Z",
          },
        },
        "topics": Array [],
        "url": "https://www.farsetlabs.org.uk/events/#upcoming-events",
        "venue": Object {
          "address": "Weavers Court, Linfield Road, BT12 5GH",
          "city": "Belfast",
          "country": "GB",
          "latitude": "54.592826",
          "longitude": "-5.940666",
          "name": "Farset Labs",
        },
      },
    ],
  ],
]
`;
